mlflow/tracing/utils/__init__.py

```diff
@@ -606,3 +607,19 @@ def _bypass_attribute_guard(span: OTelSpan) -> Generator[None, None, None]:
         yield
     finally:
         span._end_time = original_end_time
+
+
+def parse_trace_id_v4(trace_id: str) -> tuple[str | None, str]:
+    """
+    Parse the trace ID into location and trace ID components.
+    """
+    if trace_id.startswith(TRACE_ID_V4_PREFIX):
+        splits = trace_id.removeprefix(TRACE_ID_V4_PREFIX).split("/")
+        if len(splits) != 2 or splits[0] == "" or splits[1] == "":
```

Comment:
Would it also be worth validating that the second portion (`splits[1]`) is a 32-character lowercase hex character UUID here?

Reply:
Actually do we need to check here? Just since if the backend changes or something then we need client updates

Reply:
I'll merge this PR first, can update later if this is required

Reply:
@serena-ruan I think it's unlikely that the backend changes the ID to not be a 32-character hex since that is the OTEL-required format. However, I agree it's fine since we'll still validate there